File: HNRZ Salary Manager/app/Models/Bonus.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Bonus extends Model
{
    protected $fillable = [
        'nama_bonus',
        'nominal_bonus',
        'jenis_bonus',
        'periode_bonus',
        'keterangan',
    ];

    protected $casts = [
        'periode_bonus' => 'date',
        'nominal_bonus' => 'decimal:2',
    ];

    public function employees()
    {
        return $this->belongsToMany(Employee::class, 'bonus_employee');
    }

    // Accessor untuk menampilkan periode dalam format Bulan Tahun
    public function getPeriodeLabelAttribute(): string
    {
        return $this->periode_bonus->translatedFormat('F Y');
    }

    // Accessor untuk menampilkan nominal dalam format Rupiah
    public function getNominalFormatAttribute(): string
    {
        return 'Rp ' . number_format($this->nominal_bonus, 0, ',', '.');
    }
}

File: hnrz_salary_manager/lib/models/bonus.dart
class Bonus {
  final int id;
  final String namaBonus;
  final double nominalBonus;
  final String jenisBonus;
  final String periodeBonus;
  final String? keterangan;

  Bonus({
    required this.id,
    required this.namaBonus,
    required this.nominalBonus,
    required this.jenisBonus,
    required this.periodeBonus,
    this.keterangan,
  });

  factory Bonus.fromJson(Map<String, dynamic> json) {
    return Bonus(
      id: json['id'],
      namaBonus: json['nama_bonus'],
      nominalBonus: double.parse(json['nominal_bonus'].toString()),
      jenisBonus: json['jenis_bonus'],
      periodeBonus: json['periode_bonus'],
      keterangan: json['keterangan'],
    );
  }
}